### PR TITLE
Update Gluon example folder

### PR DESCRIPTION
**Pull request by ThomasDelteil**
Currently the `gluon` folder in the `example` folder is a bit of a mess. 
First step in fixing it: I moved scripts to individual folder and fixed what needed fixing.

The `image_classification.py` is remaining where it is for now because it is referenced in several places and I know some automated scripts are making use of it. I leave it to a further task to properly reorganize the `example` folder after consultation with the community.

### CODE REVIEW

**Comment by roywei** [comment]
@ThomasDelteil Thanks for the contribution! could you trigger CI again? the failing test is now temporary disabled
@mxnet-label-bot [pr-awaiting-review]

**Comment by roywei** [comment]
@mxnet-label-bot [pr-awaiting-merge,Example]

**Comment by ankkhedia** [comment]
@mxnet-label-bot [pr-awaiting-merge]

**Comment by ThomasDelteil** [comment]
I have updated the documents that refer to example/gluon

**Comment by lupesko** [comment]
@aaronmarkham does this change looks good to you? can you please review and approve or provide feedback?

**Comment by aaronmarkham** [comment]
@lupesko I just had [concerns with broken links](https://github.com/apache/incubator-mxnet/pull/12951#pullrequestreview-170934313). Because this PR will cause them (at least one), and guess who gets that CI report? Me! NBD, but it would be cool if they were addressed in advance.

**Comment by ThomasDelteil** [comment]
@aaronmarkham broken link has been addressed in the previous commit

File: docs/tutorials/unsupervised_learning/gan.md [hunk 1]
```diff
@@ -394,7 +394,7 @@ As a result, we have created two neural nets: a Generator, which is able to crea
 Along the way, we have learned how to do the image manipulation and visualization that is associated with the training of deep neural nets. We have also learned how to use MXNet's Module APIs to perform advanced model training functionality to fit the model.
 
 ## Acknowledgements
-This tutorial is based on [MXNet DCGAN codebase](https://github.com/apache/incubator-mxnet/blob/master/example/gluon/dcgan.py),
+This tutorial is based on [MXNet DCGAN codebase](https://github.com/apache/incubator-mxnet/blob/master/example/gluon/dc_gan/dcgan.py),
```
Comment (ChaiBapchya) [inline docs/tutorials/unsupervised_learning/gan.md]: ~~Getting a 404 
Here's the correct one - https://github.com/apache/incubator-mxnet/blob/master/example/gluon/DCGAN/dcgan.py~~

**Comment by ThomasDelteil** [comment]
@nswamy @indhub @sandeep-krishnamurthy @szha could you merge this please? thanks!

**Comment by ThomasDelteil** [comment]
@marcoabreu @eric-haibin-lin @hetong007 can you please merge this PR ? Thanks!

**Pull request merged**